### ISSUE TITLE
Define how object-fit and object-position interact with preserveAspectRatio

### ISSUE THREAD

**Issue by AmeliaBR** [w3c/svgwg#239]
SVG 2 has an annotation (at the end of [the `image` section](https://svgwg.org/svg2-draft/embedded.html#ImageElement)) stating:

> SVG 2 Requirement:	Support the ‘object-fit’ and ‘object-position’ properties from css3-images.

It references a general [resolution from 2011](https://www.w3.org/2011/07/29-svg-minutes.html#item08) that "SVG2 will depend on CSS3 Image Values"; these specific properties were not discussed at that meeting, however.

As I clean up the section on sizing and positioning the HTML embedded elements (e.g., `video`), I will be making it clear that `object-fit` and `object-position` apply for those elements.  However, for `image` it gets a little more complicated, since we already have `preserveAspectRatio`.

My long-term plan was to re-define `preserveAspectRatio` as a deprecated presentation hint for `object-fit` and `object-position`, as part of a larger goal to CSS-ify `viewBox` (see w3c/fxtf-drafts#7).  But that's not happening in SVG 2.  All the  `preserveAspectRatio` values can be mapped to a combination of `object-fit` and `object-position` values, but the CSS properties are much more flexible.

For now, I'm adding a warning to the `image` section of SVG 2, that a future spec may define how `preserveAspectRatio` and `object-fit` and `object-position` interact.

But I'm posting this issue so we don't forget, and as a record of a deferred SVG 2 feature request.

**Comment by AmeliaBR** [comment]
On thinking it through more carefully, I realized its actually easier than I though.  Here's the note I'm now adding to the image section:

<blockquote>
  The <a>'preserveAspectRatio'</a> calculations
  are applied <em>after</em> determining the <a>concrete object size</a>.
  If a value of <a>'object-fit'</a> is used that
  ensures that the concrete object size matches 
  the <a>intrinsic aspect ratio</a> of the embedded image
  (i.e., any value other than the default <span class="prop-value">fill</span>), 
  then the <a>'preserveAspectRatio'</a> value will have no effect.
</blockquote>

So, it's actually a very nice progressive enhancement.  If the new properties are supported, they'll be used to scale and position the image, and then there will be nothing left for pAR to do.  If they aren't, the pAR values apply. 

Once `object-fit` and `object-position` are well supported, we'll be able to deprecate `preserveAspectRatio` and encourage authors to use the CSS properties instead.

I'll close this issue when I push these commits.

**Comment by AmeliaBR** [comment]
Ack, slight correction.  We won't be able to completely deprecate `preserveAspectRatio`.  You'll still need to explicitly set `preserveAspectRatio="none"` if that's what you want.  But that's the least confusing of all pAR values, so that's not so bad.
